### DISCUSSION TITLE
What is the use of batch()?

### DISCUSSION THREAD

**Discussion by binarybehemoth** [reduxjs/react-redux#1899] [Q&A] [answered]
I am trying to explain the React-Redux [batch() function](https://react-redux.js.org/api/batch) on my [website](https://webcodingcenter.com/redux/connect()). In order to do so, I need a counterexample without using batch() so that the reader can see the difference. However, [as you can see here on CodeSandbox](https://codesandbox.io/s/react-redux-batching-njs034), I couldn't get the counterexample to work, as batching occurred even without calling batch(), ie. the number is incremented by two instead of one at a time.

**Comment by markerikson** [comment, marked as answer]
You're using React 18, which now _always_ does batching automatically.  But, React 17 and earlier did not do that:

https://github.com/reactwg/react-18/discussions/21

**Reply by 100terres** [reply]
Hi @markerikson :slightly_smiling_face: Do you think there's a way to disable the batching when using `react-redux` and `react` v18? I would like to make sure to re-render **everytime** an action is dispatched.

**Reply by markerikson** [reply]
Mmm... there's [the new `ReactDOM.flushSync()` API](https://reactjs.org/docs/react-dom.html#flushsync), which would force React to update the DOM immediately.  But, generally that's not a thing you should be reaching for very much.

What particular problem are you trying to solve there?

**Reply by 100terres** [reply]
Hi @markerikson, 

I'm maintaining a [fork](https://github.com/react-forked/dnd) of an [un-maintained project](https://github.com/atlassian/react-beautiful-dnd).

I do not have the full context behind the implementation, but it's a drag and drop library and there are a lot of actions dispatched. Some actions are dispatched one after the other and on some of these actions we get layout/style information directly on DOM elements to be able to do calculations and then dispatch another action.

It works with React 17, but with React 18 it does not behave as expected. I suspect that it's related to the new batching feature. That's why I'm trying to find a way to make sure the DOM is updated on each action dispatched. I've seen the new `flushSync` API, but I'm not sure how to make it work with react-redux.

**Reply by markerikson** [reply]
That's... a good question.

At a bare minimum, it would need to be something like:

```js
ReactDOM.flushSync(() => {
  dispatch(someAction())
})
```

However, given that React-Redux in turn does its own store setup and batch usage after the store notifies the UI, I'm not sure if that will actually do what you want it to.
